### ISSUE TITLE
Why no regularization is not used. AKA -> How to prevent overfitting in YOLOv2?

### ISSUE THREAD

**Issue by Jumabek** [AlexeyAB/darknet#476]
Hi @AlexeyAB ,
I cannot find any form of regularization on YOLOv2.
Do you have any ideas why regularization isn't used?

**Comment by AlexeyAB** [comment]
@Jumabek Hi, due to the use of batch normalization, a large set of training data and data augmentation - I think there is not a big problem of overfitting here.

* Yolo v1 used Dropout (regularization method).
* Yolo v2 uses Batch normalization (~regularization) instead of Dropout. 

**Batch normalization** reduces overfitting because it **has a slight regularization effects**. Similar to dropout, it adds some noise to each hidden layer’s activations. 


Page 2: https://arxiv.org/pdf/1612.08242v1.pdf

> Batch Normalization. Batch normalization leads to significant improvements in convergence while eliminating the need for other forms of regularization [7]. By adding batch normalization on all of the convolutional layers in YOLO we get more than 2% improvement in mAP. **Batch normalization also helps regularize the model. With batch normalization we can remove dropout** from the model without overfitting


----

There are 4 regularization methods:

* L2-regularization (to the function is added the sum of the squares of the weights)
* L1-regularization (to the function is added the sum of the weights)
* Dropout
* Use more images in your dataset


In general there are some approaches to avoid overfitting: https://en.wikipedia.org/wiki/Overfitting

> To lessen the chance of, or amount of, overfitting, several techniques are available (e.g. **model comparison, cross-validation, regularization, early stopping, pruning, Bayesian priors, or dropout**).

**Comment by Jumabek** [comment]
Thanks a lot. That is very informative for my current setting~
